src/stacks-hierarchy/Root_CarnetDetailsScreen.tsx

```diff
@@ -16,7 +23,32 @@ export function Root_CarnetDetailsScreen({navigation, route}: Props) {
   const now = useNow(2500);
   const {findFareContractByOrderId} = useTicketingState();
   const fc = findFareContractByOrderId(route?.params?.orderId);
+  const firstTravelRight = fc?.travelRights[0];
   const {t} = useTranslation();
+  const {enable_ticket_information} = useRemoteConfig();
+  const analytics = useAnalytics();
+
+  const {preassignedFareProducts} = useFirestoreConfiguration();
+  const preassignedFareProduct = findReferenceDataById(
+    preassignedFareProducts,
+    isOfFareProductRef(firstTravelRight) ? firstTravelRight.fareProductRef : '',
+  );
+
+  const ticketInfoParams = preassignedFareProduct && {
+    fareProductTypeConfigType: preassignedFareProduct?.type,
+    preassignedFareProductId: preassignedFareProduct?.id,
+  };
+
+  const onPressInfo = () => {
+    ticketInfoParams &&
+      navigation.navigate('Root_TicketInformationScreen', ticketInfoParams);
+    ticketInfoParams &&
+      analytics.logEvent(
+        'Ticketing',
+        'Ticket information button clicked',
+        ticketInfoParams,
+      );
+  };
```

Comment:
Could a reusable hook be created for this and be used both here and in `Root_FareContractDetailsScreen`?

Reply:
Done, need a re-review most likely :)